gradle/validation/missing-docs-check.gradle

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+allprojects {
+  task checkMissingDocs() {
+    group 'Verification'
+    description 'Check missing Javadocs'
+
+    dependsOn project.tasks.matching { task ->
+      task.name in [ 'checkMissingDocsPackage', 'checkMissingDocsClass', 'checkMissingDocsMethod' ]
+    }
+  }
+}
+
+configure(project(':lucene').subprojects) {
+
+  def methodLevelCheckProjects = [
+    project(':lucene:analysis:icu'),
+    project(':lucene:analysis:morfologik'),
+    project(':lucene:analysis:phonetic'),
+    project(':lucene:analysis:stempel'),
+    project(':lucene:classification'),
+    project(':lucene:demo'),
+    project(':lucene:expressions'),
+    project(':lucene:facet'),
+    project(':lucene:join'),
+    project(':lucene:memory'),
+    project(':lucene:suggest'),
+    project(':lucene:spatial3d')
+  ]
+
+  plugins.withType(JavaPlugin) {
+    if (project in methodLevelCheckProjects) {
+      // too many classes to fix overall to just enable the above to be level="method" right now,
+      // but we can prevent the modules that don't have problems from getting any worse
+      task checkMissingDocsMethod(type: CheckMissingDocsTask, dependsOn: 'javadoc') {
+        enabled = project.javadoc.destinationDir.exists()
+
+        dirs += [ project.javadoc.destinationDir.absolutePath ]
+        level = "method"
```

Comment:
You can invert the logic here: create checkMissingDocsMethod once and only conditionally set level inside (depending on project.path